test/metabase/api/dashboard_test.clj

```diff
@@ -1983,19 +2007,43 @@
                                                        (assoc  :card_id (:id card-2)))]]
           (is (= {:values          ["African" "American" "Artisan"]
                   :has_more_values false}
-                 (chain-filter-test/take-n-values 3 (mt/user-http-request :rasta :get 200 (chain-filter-values-url
-                                                                                           (:id dashboard)
-                                                                                           (:category-name param-keys)))))))))
+                 (->> (chain-filter-values-url (:id dashboard) (:category-name param-keys))
+                      (mt/user-http-request :rasta :get 200)
+                      (chain-filter-test/take-n-values 3)))))))
+
     (testing "should check perms for the Fields in question"
       (mt/with-temp-copy-of-db
         (with-chain-filter-fixtures [{:keys [dashboard param-keys]}]
           (perms/revoke-data-perms! (perms-group/all-users) (mt/id))
+
           ;; HACK: we currently 403 on chain-filter calls that require running a MBQL
           ;; but 200 on calls that we could just use the cache.
           ;; It's not ideal and we definitely need to have a consistent behavior
           (with-redefs [field-values/field-should-have-field-values? (fn [_] false)]
+            (is (= {:values ["African" "American" "Artisan"]
```

Comment:
Could you explain this change here? I know the current behavior is questionable but I'd like to know the rationale behind changing it now.

Reply:
Sure. This PR is adding the ability for users with _non-block data permissions_ to be able to view chain filter values for a dashboard that they have access to see. In this case `:rasta` can see the dashboard, therefore `:rasta` should see the chain filter values.